mdal/frmts/mdal_ugrid.cpp

```diff
@@ -45,6 +45,27 @@ std::vector<std::string> MDAL::DriverUgrid::findMeshesNames() const
   return meshesInFile;
 }
 
+std::string MDAL::DriverUgrid::buildUri( const std::string & )
+{
+  std::vector<std::string> meshNames = findMeshesNames();
+  if ( !meshNames.size() )
+  {
+    MDAL::Log::error( MDAL_Status::Err_UnknownFormat, name(), "No meshes found in file" + mFileName );
+    return std::string( "" );
+  }
+
+  std::string meshUris( "" );
+
+  for ( size_t it = 0; it < meshNames.size(); ++it )
+  {
+    meshUris += name() + ":\"" + mFileName + "\":" + meshNames.at( it );
+    if ( ( it + 1 ) < meshNames.size() ) // If this is not the last mesh in array, add separator
+      meshUris += MDAL_URI_SEPARATOR;
+  }
+
+  return meshUris;
```

Comment:
I think this can go to mdal_utils.hpp where as arguments you will pass vector or uris and it will merge them.
also maybe create function in mdal utils that take driver name, file name and optionally meshName to create uri?